mantidproject/mantid
# Improved workspace grouping in ReflectometryISISLoadAndProcess

**Description of work.**
   - Added check to group `TOF` workspaces and ignore those with x-axis in another unit.
   - Groups of groups avoided by adding individual workspaces from input group into the TOF group 
      and removing the original group.

**To test:**
1. Run `ReflectometryISISLoadAndProcess` with `TOF` workspaces
2. Repeat with other grouped/ungrouped workspaces
3. Check `TOF` workspaces grouped and non-`TOF` ignored

Fixes #26658 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
